src/adapter/graphql/tests/tests/test_gql_dataset_adapter_viersioned_file.rs

```diff
@@ -0,0 +1,1009 @@
+            &upload_token.0,
+            b"hello".len(),
+            bytes::Bytes::from_static(b"hello"),
+        )
+        .await
+        .unwrap();
+
+    // Finalize direct upload
+    let res = harness
+        .execute_authorized_query(
+            async_graphql::Request::new(indoc!(
+                r#"
+                mutation ($datasetId: DatasetID!, $uploadToken: String!) {
+                    datasets {
+                        byId(datasetId: $datasetId) {
+                            asVersionedFile {
+                                finishUploadNewVersion(uploadToken: $uploadToken) {
+                                    isSuccess
+                                    message
+                                    ... on UpdateVersionSuccess {
+                                        newVersion
+                                        contentHash
+                                        newHead
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                "#
+            ))
+            .variables(async_graphql::Variables::from_json(json!({
+                "datasetId": &did,
+                "uploadToken": upload_token,
+
+            }))),
+        )
+        .await;
+
+    assert!(res.is_ok(), "{res:#?}");
+    let upload_result = res.data.into_json().unwrap()["datasets"]["byId"]["asVersionedFile"]
+        ["finishUploadNewVersion"]
+        .clone();
+    let head_v1 = &upload_result["newHead"];
+    pretty_assertions::assert_eq!(
+        upload_result,
+        json!({
+            "isSuccess": true,
+            "message": "",
+            "newVersion": 1,
+            "contentHash": odf::Multihash::from_digest_sha3_256(b"hello"),
+            "newHead": head_v1,
+        })
+    );
+
+    // // Read back latest content
+    // let res = harness
+    //     .execute_authorized_query(
+    //         async_graphql::Request::new(indoc!(
+    //             r#"
+    //             query ($datasetId: DatasetID!) {
+    //                 datasets {
+    //                     byId(datasetId: $datasetId) {
+    //                         asVersionedFile {
+    //                             latest {
+    //                                 version
+    //                                 blockHash
+    //                                 contentType
+    //                                 contentHash
+    //                                 extraData
+    //                                 contentDownload {
+    //                                     url
+    //                                     headers { key value }
+    //                                     expiresAt
+    //                                 }
+    //                             }
+    //                         }
+    //                     }
+    //                 }
+    //             }
+    //             "#
```

Comment:
The harness is using GQL API in embedded mode so with its current configuration I cannot test the full path of direct upload/download. Will likely remove this and eventually address this via e2e test.